backend/daemon/api/entities/v1alpha/entities.go

```diff
@@ -400,6 +398,159 @@ func (api *Server) SearchEntities(ctx context.Context, in *entities.SearchEntiti
 	return &entities.SearchEntitiesResponse{Entities: matchingEntities}, nil
 }
 
+// DeleteEntity implements the corresponding gRPC method.
+func (api *Server) DeleteEntity(ctx context.Context, in *entities.DeleteEntityRequest) (*emptypb.Empty, error) {
+	var meta string
+	var qGetResourceMetadata = dqb.Str(`
+  	SELECT meta from meta_view
+	WHERE iri = :eid
+	`)
+
+	if in.Id == "" {
+		return nil, status.Errorf(codes.InvalidArgument, "must specify entity ID to delete")
+	}
+
+	eid := hyper.EntityID(in.Id)
+
+	err := api.blobs.Query(ctx, func(conn *sqlite.Conn) error {
+		return sqlitex.Exec(conn, qGetResourceMetadata(), func(stmt *sqlite.Stmt) error {
+			meta = stmt.ColumnText(0)
+			return nil
+		}, in.Id)
+	})
+	if err != nil {
+		return nil, err
+	}
+	_, err = &emptypb.Empty{}, api.blobs.Query(ctx, func(conn *sqlite.Conn) error {
+		return api.blobs.ForEachComment(ctx, eid.String(), func(c cid.Cid, cmt hyper.Comment) error {
+			referencedDocument := strings.Split(cmt.Target, "?v=")[0]
+			if referencedDocument == eid.String() {
+				var qEmptyComments = dqb.Str(`
+					UPDATE blobs
+					SET data = NULL, size = -1
+					WHERE multihash = :hash
+				`)
+				_, err = hypersql.BlobsDelete(conn, c.Hash())
+				if err != nil {
+					_, err = &emptypb.Empty{}, api.blobs.Query(ctx, func(conn *sqlite.Conn) error {
+						return sqlitex.Exec(conn, qEmptyComments(), func(_ *sqlite.Stmt) error {
+							return nil
+						}, c.Hash())
+					})
+					if err != nil {
+						return err
+					}
```

Comment:
Looks like this might be the timeout issue that we have in CI, because it tries to acquire a DB connection while already having one. So maybe it deadlocks there because there's no more free connections in the pool. Not sure how much CPU cores do CI machines have.